supportedBackends/jes/src/main/scala/cromwell/backend/impl/jes/RunStatus.scala

```diff
@@ -51,12 +51,16 @@ object RunStatus {
               zone: Option[String],
               instanceName: Option[String]): UnsuccessfulRunStatus = {
       val jesCode: Option[Int] = errorMessage flatMap { em => Try(em.substring(0, em.indexOf(':')).toInt).toOption }
-      if (errorCode == Status.ABORTED && jesCode.contains(JesAsyncBackendJobExecutionActor.JesPreemption)) {
-        Preempted(errorCode, jesCode, errorMessage, eventList, machineType, zone, instanceName)
-      } else if (errorCode == Status.CANCELLED) {
-        Cancelled(errorCode, jesCode, errorMessage, eventList, machineType, zone, instanceName)
-      } else {
-        Failed(errorCode, jesCode, errorMessage, eventList, machineType, zone, instanceName)
+
+      errorCode match {
+        case Status.ABORTED if jesCode.contains(JesAsyncBackendJobExecutionActor.JesPreemption) =>
+          Preempted(errorCode, jesCode, errorMessage, eventList, machineType, zone, instanceName)
+        case Status.UNKNOWN if errorMessage.exists(_.contains("failed to start due to preemption")) =>
+          Preempted(errorCode, jesCode, errorMessage, eventList, machineType, zone, instanceName)
+        case Status.CANCELLED =>
+          Cancelled(errorCode, jesCode, errorMessage, eventList, machineType, zone, instanceName)
+        case _ =>
+          Failed(errorCode, jesCode, errorMessage, eventList, machineType, zone, instanceName)
       }
```

Comment:
Even DRYer, arguably more readable:
```scala
val unsuccessfulStatusBuilder = errorCode match {
  case Status.ABORTED if jesCode.contains(JesAsyncBackendJobExecutionActor.JesPreemption) => Preempted
  case Status.UNKNOWN if errorMessage.exists(_.contains("failed to start due to preemption")) => Preempted
  case Status.CANCELLED => Cancelled
  case _ => Failed
}
unsuccessfulStatusBuilder.apply(errorCode, jesCode, errorMessage, eventList, machineType, zone, instanceName)
```

Reply:
Is this an acceptable compromise? I find it more obvious what's happening:
```scala
case Status.CANCELLED => Cancelled.apply _
```